warcraft/dapp/src/components/payment-widget.js

```diff
@@ -0,0 +1,156 @@
+/* global AlgoSigner */
+import "./signer.css";
+
+import { types, WebMode } from "@algo-builder/web";
+import { Button, CircularProgress, Typography } from "@material-ui/core";
+import PropTypes from "prop-types";
+import React from "react";
+import { useCallback, useState } from "react";
+
+import { CHAIN_NAME } from "../algosigner.config";
+
+const CONFIRMED_ROUND = "confirmed-round";
+
+/**
+ * Get default account address from user's wallet. Returns undefined otherwise.
+ * NOTE: Currently this returns the first address from list of account addresses
+ *
+ * TODO: Update this function to return "current active" account in wallet after
+ * https://github.com/PureStake/algosigner/issues/252 is resolved.
+ * Task - https://www.pivotaltracker.com/story/show/178760753
+ */
+async function getDefaultAccountAddr() {
+  const walletAccounts =
+    (await AlgoSigner.accounts({
+      ledger: CHAIN_NAME,
+    })) ?? [];
+
+  if (walletAccounts.length) {
+    return walletAccounts[0].address;
+  }
+  return undefined;
+}
+
+/**
+ * Executes transaction to receive payment of ticket
+ * @param fromAddress address of currently active account in wallet
+ * @param toAddress address to recieve payment (currently set to master-account)
+ * @param amount amount (in ALGOs)
+ * @param setLoading setLoading function to set loading state in button
+ * @returns response of transaction OR rejection message
+ */
+async function executePayment(fromAddress, toAddress, appIndex, setLoading) {
+  try {
+    const web = new WebMode(AlgoSigner, CHAIN_NAME);
+    const groupTx = [
+      {
+        type: types.TransactionType.TransferAsset,
+        sign: types.SignType.SecretKey,
+        fromAccountAddr: fromAddress,
+        toAccountAddr: toAddress,
+        amount: 1,
+        payFlags: {},
+      },
+      {
+        type: types.TransactionType.OptInToApp,
+        sign: types.SignType.SecretKey,
+        fromAccountAddr: fromAddress,
+        toAccountAddr: toAddress,
+        appID: appIndex,
+        payFlags: {},
+      },
+    ];
+    // show loading state on button while we send & wait for transaction response
+    setLoading(true);
+    let response = await web.executeTransaction(groupTx);
+    console.log(response);
+    const confirmedTxInfo = {
+      txId: response.txId,
+      confirmedRound: response[CONFIRMED_ROUND],
+    };
+
+    return [
+      <pre key={""}>
+        {"Successfully bought ticket: " +
```

Comment:
Shouldn't this be `Succesfully deposited Coin: <coin_id> & registered user <user>`?

Reply:
yes